### ISSUE TITLE
[dotnet-sdk-9.0.100-preview.2.24113.6] SharpScss.dll causes application crash when converting scss to css

### ISSUE THREAD

**Issue by Junjun-zhao** [xoofx/SharpScss#21]
When testing 3rd party application which referring to the SharpScss nuget package with the latest .NET 9 build, it stops working after accessing page. It is caused by SharpScss.dll when converting scss to css. The exception from Visual Studio output window is: "Exception thrown: 'SharpScss.ScssException' in SharpScss.dll".  Please refer to https://github.com/dotnet/runtime/issues/98646 for more reference.

**Reproduction Steps**
Minimal Repro steps (Demo attached):
[ConsoleDemo.zip](https://github.com/dotnet/runtime/files/14327789/ConsoleDemo.zip)

The machine has dotnet-sdk-9.0.100-preview.2.24113.6 installed.

1. Create a new console project.
2. Install SharpScss nuget package.
3. Copy the following code to Program.cs:
```
using SharpScss;
var result = Scss.ConvertToCss(@"@import ""foo"";", new ScssOptions()
{
    OutputStyle = ScssOutputStyle.Compressed,
    InputFile = "test.scss",
    TryImport = (ref string file, string path, out string scss, out string map) =>
    {
        scss = "div {color: #FFF;}";
        map = null;
        return true;
    }
});
var css = result.Css.Trim();
Console.WriteLine(css);
Console.ReadKey();
```
4. Build the app.
5. Change the runtime.config file to let the app run against with dotnet-sdk-9.0.100-preview.2.24113.6.
```
 "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "9.0.0-preview.2.24112.33"
    }
```
6. Launch the app.

**Expected behavior**
Launch successfully with converted css output in command window:
![image](https://github.com/xoofx/SharpScss/assets/83067197/71db8be1-c974-4809-bd3e-d904efa3a325)

**Actual behavior**
App crashed with stop working prompt window.

**Comment by ronikurnia1** [comment]
Hi @Junjun-zhao, I've got the same issue. Do you have any workaround?

**Comment by Junjun-zhao** [comment]
Hi @ronikurnia1 No, we are still blocked by this issue.

**Comment by xoofx** [comment]
`2.1.0` should fix the issue, but while investigating, I believe it is a regression in .NET 9 and DllImport interop/trampoline passing an invalid argument when passing null but haven't been able to reproduce it with a simpler example.

**Comment by ronikurnia1** [comment]
I've just tested it using `2.1.0` and it works like a charm in my env. Great job thanks @xoofx !

**Comment by xoofx** [comment]
Actually, I think it is tracked in this issue https://github.com/dotnet/runtime/issues/109033 and it should be fixed in a upcoming version of .NET 9

**Comment by Junjun-zhao** [comment]
Thanks @xoofx. This issue has been fixed on the latest 3.0.0 and .NET 9 GA build.
